### DISCUSSION TITLE
Theming: disabled button icon

### DISCUSSION THREAD

**Discussion by ConanGalloway** [react-native-elements/react-native-elements#3327] [Q&A]
I'm trying to see if there's a way in a theme to specify the color for a button icon when disabled?  For example I might have

```
Button: {
  buttonStyle: {
    raised: true,
    borderRadius: 15,
    backgroundColor: '#440e62',
  },
  titleStyle: {
    color: '#ded000'
  },
  icon:{
    color:'#ded000',
  },
  disabledStyle:{
    backgroundColor:'#dddddd'
  },
  type: 'clear'
},
```

The icon color will match the text when active but when disabled the text falls back to the default grey while the icons stays yellow.  The icon docs specify that you can set a disabledStyle and I've tried embedding that in the icon member above but I think I'm missing something.  Any suggestion would be appreciated.

**Comment by arpitBhalla** [comment]


**Reply by Nantris** [reply]
@arpitBhalla is this addressed yet? And is there a plan to do the same for the base `<Icon />`, I assume?

**Reply by arpitBhalla** [reply]
I believe it would work

**Reply by Nantris** [reply]
Whoops I misread what you meant.

I guess my question is, why can't we set the the icon color via `disabledStyle.color`? That would be a lot more in keeping with the API that React Native Elements has set out for all their components, and this seems to be a weird exception. The `props => ...` approach is a fine workaround, but the API is confusing and I wouldn't think it would be necessary - and that `disabledStyle.color` would override `color` when it's defined.

**Reply by arpitBhalla** [reply]
I think `disabledStyle` is for disabled state of `Button`, and here we want to change `Icon`'s color when button is disabled.

**Reply by ConanGalloway** [reply]
It does seem a little inconsistent but exposing the disabled prop and using a ternary does address the issue.

`Button: (buttonProps) => ({  
buttonStyle: {  
raised: true, 
    borderRadius: 15,  
    backgroundColor: '#440e62',  
},
  titleStyle: {
  color: '#ded000'
},
  disabledStyle: {
  backgroundColor: '#dddddd,
},
  icon: {
      color: buttonProps.disabled  ? '#eeeeee' : '#ded000',
  },
  type: 'clear'
}),`

EDIT: I'm adding the 2x spaces at the end of my code lines but they aren't causing line returns.  If you review the code from my OP you'll get the gist.

**Reply by Nantris** [reply]
But it's not just for Buttons, because Icons also have a `disabled` prop and `disabledStyles` also. 

@arpitBhalla the issue I'm talking about is what @ConanGalloway just provided code for working around. But my question is why we need to access `buttonProps.disabled` or `iconProps.disabled` and use a ternary to set a `color` for the `disabled` state, when I'd expect to be able to set `disabledStyles.color` and have that work.
